solutions/usaco-596.mdx

```diff
@@ -97,3 +97,73 @@ void ff(int i, int j)
 </CPPSection>
 
 </LanguageSection>
+	
+## Alternate Faster Solution
+
+It is possible to make the following 2 observations:
+
+1. For each contiguous closed region, there must be 1 gate. In a minimum state, each region has exactly 1 gate.
+2. Each time FJ comes across a node he has visited along an edge he has not crossed in either direction, he creates exactly 1 new region. In other words, FJ creates exactly 1 new contiguous closed region if all of these conditions are met:
+	* FJ walks along an edge from point $a$ to $b$
+	* He did not walk from $a$ to $b$ in the past
+	* He did not walk from $b$ to $a$ in the past
+	* $b$ is a point he had already visited
+
+Under these observations, we can use a set of visited edges and visited nodes, and keep track of the number of contiguous closed regions. This solution is almost 10 times faster, and works in $\mathcal{O}(N\log N)$ due to $\log N$ set lookup and insertion. 
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
```

Comment:
move to line 102

Reply:
oops i think i commented on the wrong comment but https://github.com/cpinitiative/usaco-guide/pull/1828#discussion_r659398168

Reply:
@maggie-j-liu told me to move it down there